### PR TITLE
Update writing_your_start-up_command.md

### PR DESCRIPTION
**Pull request by ximae**
changed the folder scanning script so it can use folder names with spaces and some illegal characters. 
Also added a line so it checks that the files being made a crate are actually folders.

### CODE REVIEW

File: writing_your_start-up_command.md [hunk 1]
```diff
@@ -80,20 +80,28 @@ Explain. Also explain you can assign an alias to the script itself. FIXME
 
 Each directory within music will show up as a crate inside xwax, etc. FIXME
 
+(Fixed so it can take folders names with spaces and some illegal characters.)
```
Comment (lpil) [inline writing_your_start-up_command.md]: No need for this remark I think :)

File: writing_your_start-up_command.md [hunk 2]
```diff
@@ -80,20 +80,28 @@ Explain. Also explain you can assign an alias to the script itself. FIXME
...
+
 Should this be it's own page?
 
 ```bash
-#!/bin/sh
+#!/bin/bash
```
Comment (lpil) [inline writing_your_start-up_command.md]:
```
#!/usr/bin/env bash
```

File: writing_your_start-up_command.md [hunk 3]
```diff
@@ -80,20 +80,28 @@ Explain. Also explain you can assign an alias to the script itself. FIXME
...
 
 # The loop
 # Replace ~/Music/* with the location of the directories you wish to scan FIXME
+
 PLAYLISTS=
-for P in ~/Music/*; do
-    PLAYLISTS="-l $P $PLAYLISTS"
+  for P in ~/Music/*; do
+    if [ -d "$P" ]; then
+     PLAYLISTS="-l \"$P\" $PLAYLISTS"
```
Comment (lpil) [inline writing_your_start-up_command.md]: Funky indentation here

File: writing_your_start-up_command.md [hunk 4]
```diff
@@ -80,20 +80,28 @@ Explain. Also explain you can assign an alias to the script itself. FIXME
...
+    fi
 done
 
-# Replace this start up command with your own, except keeping $PLAYLISTS in place of the -l flag
-xwax -a plughw:Audio4DJ,0,0 -a plughw:Audio4DJ,0,1 $PLAYLISTS
+# Replace this start up command with your own, except keeping $(echo $PLAYLISTS) in place of the -l flag 
+# and changing the terminal to be used, for example lxterminal instead of gnome-terminal.
+
+gnome-terminal -e "xwax -a plughw:Audio4DJ,0,0 -a plughw:Audio4DJ,0,1 $(echo $PLAYLISTS)"
```
Comment (lpil) [inline writing_your_start-up_command.md]: I'd prefer to avoid the gnome-terminal dep here.

**Review by lpil — CHANGES_REQUESTED** [review]

File: writing_your_start-up_command.md (hunk 4 above)
Comment (ximae) [reply writing_your_start-up_command.md]: it needs to be launched through a new terminal or it wont work for some funky reason.

doesnt have to be gnome-terminal thugh, can be any.

File: writing_your_start-up_command.md (hunk 1 above)
Comment (ximae) [reply writing_your_start-up_command.md]: ur right, first time i do a pull request didnt know i could leave comments of the changes afterwards. :D

File: writing_your_start-up_command.md (hunk 4 above)
Comment (lpil) [reply writing_your_start-up_command.md]: This should work without a new terminal. What happens?

Comment (ximae) [reply writing_your_start-up_command.md]: for some reason it doesnt take the double quotes in the path with a starightforward xwax command through the script ( thus not taking spaces in the foldername), but it does if manually add the double quotes to the path without the script or through a new terminal in the script.

its some odd behaviour i dont understand.

I opened a thread about it in the mailing list a week or two ago.  ill link you the message.

https://sourceforge.net/p/xwax/mailman/message/35504960/

just go ahead and try the script with a straightforward xwax command and then calling xwax through a new terminal.

**Comment by lpil** [comment]
Thanks!

**Review by lpil — APPROVED** [review]

**Comment by lpil** [comment]
Sorry about the incredible delay here, I did not see the updates

**Pull request merged**